### PR TITLE
 Refactor pih.config to be a global property

### PR DESCRIPTION
**Pull request by brandones**
This takes advantage of https://github.com/openmrs/openmrs-sdk/pull/73 to allow pih.config and pih.config.dir to be set at setup time. This requires turning those properties from runtime properties into global properties.

### CODE REVIEW

**Comment by brandones** [comment]
Can we have a call about this sometime soon, or do whatever we need to do to make a decision to merge it or to close it?

**Comment by mogoodrich** [comment]
Yeah, sorry, I still have a to do on this but it kind of got pushed to the bottom of the pile, let's reassess next week.

**Comment by mseaton** [comment]
Happy to discuss at your convenience @brandones .  My intuition continue to be that this is better done in something other than global properties.  I'm thinking that we'd want to build into the SDK a mechanism to more easily insert custom setup logic.  In general I'd prefer adapting the SDK to match the way we want our code configured/deployed, then to bend our code for what is currently possible in a dev environment setup.

**Comment by brandones** [comment]
What are your concerns with them being global properties?

**Comment by mseaton** [comment]
Generally I think of global properties as settings that can be configured and changed at runtime, whereas the pih config is something that really can't be changed after the first time the server is run, as it fundamentally controls how the system is set up, what metadata is installed, etc.  Maybe it doesn't matter or I'm wrong, but it just feels like a different class of configuration.

**Comment by brandones** [comment]
That does make sense. Our choice then is to either accept this as a hack to simplify our set-up, or file a ticket against OpenMRS to support setting runtime properties during set-up (and probably eventually doing that ticket, if it's going to get done).

**Comment by brandones** [comment]
I'm gonna go ahead and close this. We can resurrect it from the branch someday if we're interested.

**Comment by mogoodrich** [comment]
Fair enough... I hadn't 100% given up on getting the PR, but I probably wasn't being honest with myself... :)

File: api/src/main/resources/openmrs-distro.properties [hunk 1]
```diff
@@ -52,5 +52,9 @@ omod.orderentryui=${orderentryuiVersion}
 owa.openmrs-owa-cohortbuilder=${cohortBuilderOwaVersion}
 omod.mirebalais=${project.parent.version}
 db.h2.supported=false
+property.pih.config.prompt=Please specify the PIH config files to use
+property.pih.config.default=mirebalais,mirebalais-humci
+property.pih.config.dir.prompt=Please provide the path to the PIH config files
+property.pih.config.dir.default=/etc/puppet/mirebalais-modules/openmrs/files/config/
```
Comment (mseaton) [inline api/src/main/resources/openmrs-distro.properties:58]: We con't need this config directory anymore, these are now just in .OpenMRS/configuration.

File: README.md [hunk 2]
```diff
@@ -130,6 +130,10 @@ $ mvn openmrs-sdk:setup -DserverId=[serverId] -Ddistro=org.openmrs.module:mireba
 
 * When prompted, set the port to debug on (standard is 1044)
 
+* When prompted, specify the config files you'd like to use (e.g. `mirebalais,mirebalais-humci`)
+
+* When prompted, specify the path to the directory where the config files are. Something like `/[path to]/mirebalais-puppet/mirebalais-modules/openmrs/files/config`
```
Comment (mseaton) [inline README.md:135]: Remove this readme segment about the config dir

**Comment by mseaton** [comment]
This was implemented in a different PR.  Thanks @brandones !

**Pull request closed without merging**